src/services/FetchMovies.php

```diff
@@ -32,31 +36,87 @@ protected function jsonStringToArray($json_string)
     }
 
     /**
+     * @return string
+     */
+    public function getTimeZone()
+    {
+        $tz = 'America/New_York';
+        $timestamp = time();
+        $dt = new \DateTime("now", new \DateTimeZone($tz)); //first argument "must" be a string
+        $dt->setTimestamp($timestamp); //adjust the object to correct timestamp
+        return $dt->format('Y-m-d');
+    }
+
+    /**
      * @return bool|string
      */
     protected function getJsonString()
     {
         $completeurl = $this->nowPlaying_url . $this->api_key;
         $contents = file_get_contents($completeurl);
+        return $contents;
+    }
+
+    /**
+     * @param array $raw_movies_array
+     * @return array
+     */
+    protected function toMovieType(array $raw_movies_array)
+    {
+        //for each loop required
+        $movie_array = [];
+        foreach ($raw_movies_array as $raw_movie_array) {
+            $movie = new Movie();
+            $movie->id = $raw_movie_array['id'];
+            $movie->title = $raw_movie_array['title'];
+            $movie->release_date = $raw_movie_array['release_date'];
+            $movie->poster_image = $raw_movie_array['poster_path'];
+            $movie->critic_rating = $raw_movie_array['vote_average'];
+            $movie->description = $raw_movie_array['overview'];
+            $movie_array[] = $movie;
+        }
+
+        return $movie_array;
+    }
 
+    /**
+     * @return array
+     */
+    public function getUpcomingMovies()
+    {
+        $data_string = $this->getUpcomingString();
+        $data_arr = $this->jsonStringToArray($data_string);
+        //need to pass through array converting to movie types
+        $movie_data = $this->toMovieType($data_arr['results']);
+        return $movie_data;
+    }
+
+    /**
+     * @return bool|string
+     */
+    protected function getUpcomingString()
+    {
+        $date = $this->getTimeZone();
+        $params = [
+            'api_key' => $this->api_key,
```

Comment:
I would move the setting of the api key and language to the buildApiUrl method since all calls to the api will need it.

Reply:
They api key needs to be the first param passed in or else it messes with the url. How do I do this when I am initializing params in the getUpcomingString method?